common/page.go

```diff
@@ -364,6 +363,105 @@ func (p *Page) initEvents() {
 	}()
 }
 
+// urlTagName is used to check the incoming metric url tag against user
+// defined url regexes. When a match is found a user defined name, which is to
+// be used in the urls place in the url metric tag, is returned.
+//
+// The check is done by calling the handlers that were registered with
+// `page.on('metric')`. The user will need to use `Tag` to supply the
+// url regexes and the matching is done from within there. If a match is found,
+// the supplied name is returned back upstream to the caller of urlTagName.
+func (p *Page) urlTagName(ctx context.Context, urlTag string) (string, bool) {
+	p.eventHandlersMu.RLock()
+
+	// If there are no handlers for EventConsoleAPICalled.
+	if _, ok := p.eventHandlers[EventPageMetricCalled]; !ok {
+		p.eventHandlersMu.RUnlock()
+
+		return "", false
+	}
```

Comment:
We might put this in a different function so that the lock handling would be simpler. We could also call that function from `Page.onConsoleAPICalled`. This way we won't need to provide `RUnlock` in the loop as a side benefit.

Reply:
Yep, you've mentioned this earlier. It's in the generalize issue 👍